### PR TITLE
Normalizing flows

### PR DESCRIPTION
**Pull request by shulgin-egor**
Planar and Radial Flows implemented. Notebook with experiments will be committed soon.

### CODE REVIEW

File: src/DeepGenerativeModels/AutoEncoders.py [hunk 1]
```diff
@@ -63,8 +61,8 @@ def p_z(self, num_samples):
             mu,    Tensor - the matrix of shape num_samples x latent_dim.
             sigma, Tensor - the matrix of shape num_samples x latent_dim.
         """
-        mu = torch.zeros([num_samples, self.latent_dim], device=self.device)
```
Comment (r-isachenko) [inline src/DeepGenerativeModels/AutoEncoders.py:66]: полагаю, что ты не обновил файлы до последней версии

File: src/DeepGenerativeModels/NormalizingFlow.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        
+        Args: dim (int): The dimension of input space (data dimensionality).
+        """
+        init.uniform_(self.w, -math.sqrt(1/dim), math.sqrt(1/dim))
+        init.uniform_(self.u, -math.sqrt(1/dim), math.sqrt(1/dim))
+        init.uniform_(self.b, -math.sqrt(1/dim), math.sqrt(1/dim))
+
+    def forward(self, x):
+        """
+        Applies a Normalizing Flow (mapping from data `x` to the noise `z`).
```
Comment (r-isachenko) [inline src/DeepGenerativeModels/NormalizingFlow.py:92]: если ты вдруг откуда то дернул этот код, то вставь, пожалуйста, ссылку на источник

**Review by r-isachenko — CHANGES_REQUESTED** [review]
VAE_IWAE.ipynb, полагаю, ты не собирался коммитить
не хватает твоего ноутбука

File: src/DeepGenerativeModels/NormalizingFlow.py (hunk 2 above)
Comment (shulgin-egor) [reply src/DeepGenerativeModels/NormalizingFlow.py:92]: Я указал это в начале файла NormalizingFlows.py.
